src/core/json_utils.rs

```diff
@@ -90,20 +90,20 @@ pub(crate) fn index_json_values<'a>(
     Ok(())
 }
 
-fn index_json_object(
+fn index_json_object<'a, V: JsonVisitor<'a>>(
     doc: DocId,
-    json_value: &serde_json::Map<String, serde_json::Value>,
+    mut json_visitor: V,
```

Comment:
not sure with the lifetime, but can we write this as `impl JsonVisitor`

Reply:
I think so, but I don't think we really gain anything having it be an anonymous type rather than a regular generic

Reply:
I think in library interface, impl trait in argument position is a bit of an anti-pattern compared to regular generics. The main problem is that user code cannot specific the type using the turbofish syntax. This can be worked through casts, extra bindings with type ascriptions, etc., but as API courtesy, named generic parameter seem preferable.